addons/website/static/src/components/dialog/page_properties.js

```diff
@@ -135,17 +135,38 @@ export class DuplicatePageDialog extends Component {
     }
 
     async duplicate() {
+        const duplicates = [];
         if (this.state.name) {
-            const res = await this.orm.call(
-                'website.page',
-                'clone_page',
-                [this.props.pageId, this.state.name]
-            );
-            this.website.goToWebsite({path: res, edition: true});
+            for (let count = 0; count < this.props.pageIds.length; count++) {
+                const name = this.state.name + (count ? ` ${count + 1}` : "");
+                duplicates.push(await this.orm.call(
+                    'website.page',
+                    'clone_page',
+                    [this.props.pageIds[count], name]
+                ));
+            }
         }
-        this.props.onDuplicate();
+        this.props.onDuplicate(duplicates);
     }
 }
+DuplicatePageDialog.components = {WebsiteDialog};
+DuplicatePageDialog.template = xml`
+<WebsiteDialog close="props.close" primaryClick="() => this.duplicate()">
+    <div class="mb-3 row">
+        <label class="col-form-label col-md-3">
+            Page Name
+        </label>
+        <div class="col-md-9">
+            <input type="text" t-model="state.name" class="form-control" required="required" t-ref="autofocus"/>
+        </div>
+    </div>
+</WebsiteDialog>
+`;
+DuplicatePageDialog.props = {
+    onDuplicate: {type: Function, optional: true},
+    close: Function,
+    pageIds: { type: Array, element: Number, optional: true },
+};
```

Comment:
Not sure to understand all this, this is already defined just above? Probably just a rebase/forward port unseen error?
```js
export class DuplicatePageDialog extends Component {
    static components = { WebsiteDialog };
    static template = xml`
    <WebsiteDialog close="props.close" primaryClick="() => this.duplicate()">
        <div class="mb-3 row">
            <label class="col-form-label col-md-3">
                Page Name
            </label>
            <div class="col-md-9">
                <input type="text" t-model="state.name" class="form-control" required="required" t-ref="autofocus"/>
            </div>
        </div>
    </WebsiteDialog>
    `;
    static props = {
        onDuplicate: Function,
        close: Function,
        pageId: Number,
    };
```

In the now preferred way to declare those props and template since https://github.com/odoo/odoo/commit/06891ab6e0bf3de7e136bf136923bd148dbb6c1d

Also note that I made https://github.com/odoo/odoo/commit/e7396be84e38934b8bf78169d1f4e28420421445 a few days ago to correct the `pageId` property, which hsouldn't be marked as optional